python/sglang/srt/managers/data_parallel_controller.py

```diff
@@ -75,9 +107,26 @@ def __init__(self, server_args, port_args) -> None:
         dispatch_lookup = {
             LoadBalanceMethod.ROUND_ROBIN: self.round_robin_scheduler,
             LoadBalanceMethod.SHORTEST_QUEUE: self.shortest_queue_scheduler,
+            LoadBalanceMethod.RESOURCES_AWARE: self.resources_aware_scheduler,
+            LoadBalanceMethod.PRE_RADIX: self.pre_radix_scheduler,
         }
         self.dispatching = dispatch_lookup[self.load_balance_method]
 
+        # For resources aware
+        self.dp_size = server_args.dp_size
+        self.controller_info = ControllerInfo(server_args.dp_size)
+        self.pre_available_kv_cache = []
+        self.main_available_kv_cache = []
+
+        self.pre_num_running_req = []
+        self.main_num_running_req = []
+
+        self.pre_num_waiting_req = []
+        self.main_num_waiting_req = []
```

Comment:
Our methods uses three indicators, namely `available_mem, num_reqs_running` and `num_reqs_waiting`, which are updated by each service node. Since the information is not up to date when we schedule, we need a mechanism to ensure that the information we get is correct.
For example, when scheduling the first request, the available_mem we get from the service node is as follows: `available_mem = [30, 10]`
At this time, we will send the request to the [0]th node because it has a larger remaining memory.
If the second scheduling request is received immediately at this time, if we still collect information from each service node, we may still get `available_mem = [30, 10]`, because the service node 0 has not updated the latest remaining memory.
Therefore, we maintain `pre`, which is used to determine whether the service node has updated the corresponding indicator.
We maintain `main`, which is used as an indicator that can be used to schedule requests.
But the variable name is indeed confusing. Do you have any better suggestions?